pkg/ipcache/cell.go

```diff
@@ -0,0 +1,5 @@
+package ipcache
+
+import "github.com/cilium/cilium/pkg/hive/cell"
+
+var Cell = cell.Provide(NewIPCache)
```

Comment:
Consider extracting the interface here. `go doc ipcache.IPCache` gives a good starting point. Also see if you can split the interface up by functionality (e.g. one for lookups, one for modifying etc.).

Reply:
I gave this a shot, we can significantly simplify the public API of the ipcache but it results in quite a significant diff, so I would prefer to do that in a separate PR.